home3.1/math/src/main/java/org/math/Math.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
+ */
+
+package org.math;
+
+/**
+ *
+ * @author ubuntu
+ */
+public class Math {
+    
+    // Function to calculate the inverse value
+    public static double calculateInverse(double x, double y, double z) {
+        double product = x * y * z;
+        double sum = x + y + z;
```

Comment:
There is no need to calculate sum in prior cause for almost all the cases product will not be 0. In this situation it has small impact on performance but in general if you have some intensive task such unneeded extra calls may cause an issue.
Also as we return each time then we can write
```
double product = x * y * z;
if (product != 0) {
    return 1 / product; // Return inverse of product if it exists
} 

double sum = x + y + z;
if (sum != 0) {
     return 1 / sum; // Return inverse of sum if product is not valid
}

return x + (y + 1) * (z - 1); // Return the default value if neither product nor sum is valid


```